src/computability/reduce.lean

```diff
@@ -258,108 +255,165 @@ theorem disjoin_le {α β γ} [primcodable α] [primcodable β] [primcodable γ]
   one_one_reducible.disjoin_right.to_many_one.trans h⟩,
  λ ⟨h₁, h₂⟩, disjoin_many_one_reducible h₁ h₂⟩
 
+/-- A many-one degree is an equivalence class of sets up to many-one equivalence. -/
+def many_one_degree : Type :=
+quotient $ show setoid (Σ (A : set ℕ) [primcodable A], set A), from
```

Comment:
I would prefer for this to just be a setoid on `set nat`. There is a well defined `set nat` corresponding to any `set A` here `A` is a primcodable type, given by taking the image by `encode`, and that can define what the degree of the set is.